src/gui/base/Overlay.ts

```diff
@@ -104,19 +96,21 @@ export const overlay: Component = {
 							height: position.height,
 							"z-index": position.zIndex != null ? position.zIndex : LayerType.Overlay,
 						},
-						onanimationend: () => {
-							if (attrs.isClosing) {
-								overlays.delete(key)
-							}
-						},
-						onupdate(vnode: VnodeDOM<any>): any {
-							if (attrs.isClosing && attrs.closeAnimation != null) {
+						onbeforeremove: (vnode: VnodeDOM) => {
+							if (attrs.closeAnimation != null) {
 								const dom = vnode.dom as HTMLElement
 
 								// Force the environment to restart the animations via a reflow
 								dom.className = baseClasses
 								void dom.offsetWidth
-								dom.className = classes
+
+								// Play the closing animation
+								dom.className = baseClasses + " " + attrs.closeAnimation
+
+								// Wait for the close animation to complete
+								return new Promise(function (resolve) {
```

Comment:
small nicpick but this could be an arrow function

Reply:
Good catch! Looks nicer that way.